src/main/java/net/dries007/tfc/api/capability/metal/CapabilityMetalItem.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Work under Copyright. Licensed under the EUPL.
+ * See the project README.md and LICENSE.txt for more information.
+ */
+
+package net.dries007.tfc.api.capability.metal;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+import net.minecraft.item.ItemBlock;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.CapabilityInject;
+import net.minecraftforge.common.capabilities.CapabilityManager;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.oredict.OreDictionary;
+
+import net.dries007.tfc.api.capability.DumbStorage;
+import net.dries007.tfc.api.registries.TFCRegistries;
+import net.dries007.tfc.api.types.Metal;
+import net.dries007.tfc.api.util.TFCConstants;
+import net.dries007.tfc.objects.inventory.ingredient.IIngredient;
+import net.dries007.tfc.util.Helpers;
+
+public final class CapabilityMetalItem
+{
+    @CapabilityInject(IMetalItem.class)
+    public static final Capability<IMetalItem> METAL_OBJECT_CAPABILITY = Helpers.getNull();
+    public static final ResourceLocation KEY = new ResourceLocation(TFCConstants.MOD_ID, "metal_object");
+
+    public static final Map<IIngredient<ItemStack>, Supplier<ICapabilityProvider>> CUSTOM_METAL_ITEMS = new HashMap<>(); //Used inside CT, set custom IMetalItem for items outside TFC
+    public static final Map<String, Metal.ItemType> ORE_DICT_METAL_ITEMS = new HashMap<>();
```

Comment:
Use a `LinkedHashMap<>` instead, (preserves insertion order as iteration order), Move `ingotDouble` before `ingot` in the registration, and then you simplify the checking code to not need to keep checking if one has been found. (Since hash map doesn't ensure iteration order is constant, so this may still break.